x-pack/plugins/security_solution/public/management/pages/policy/view/policy_details.tsx

```diff
@@ -128,6 +151,14 @@ export const PolicyDetails = React.memo(() => {
     setShowConfirm(false);
   }, []);
 
+  const handleShowAdvancedPolicyClick = useCallback(() => {
```

Comment:
you could collapse these into one function `handleAdvancedPolicyClick()` and simply pass the current opposite value of `showAdvancedPolicy`.

For instance:

```js
const handleAdvancedPolicyClick = useCallback(() => {
    setShowAdvancedPolicy(!showAdvancedPolicy);
  }, []);
```

Then below, you could collapse the components down to something like this:

```js
        (
          <EuiButtonEmpty
            data-test-subj="advancedPolicyButton"
            onClick={handleAdvancedPolicyClick}
          >
            <FormattedMessage
              id="xpack.securitySolution.endpoint.policy.advanced.hide"
              defaultMessage="{status} advanced settings"
              values={{
                  status: showAdvancedPolicy ? 'Hide' : 'Show',
              }}
            />
          </EuiButtonEmpty>
        )
```

Reply:
Thanks @kevinlog. When I do it this way, the button first shows up as "Show adv..". When I click on it, it shows advanced settings and switches to "Hide adv..", as expected. But then the button stops working; clicking on it doesn't hide the settings or switch the button back to "Show adv..". Any thoughts on what I might be missing? Is the change in L94 correct?

Reply:
Closed the loop with @jmiller263 offline.  We just needed to add the value to the memoization array.

```js
const handleAdvancedPolicyClick = useCallback(() => {
    setShowAdvancedPolicy(!showAdvancedPolicy);
  }, []);
```

Reply:
f44601c417470f867e7aff2f38871e449f948725